smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -49,12 +49,15 @@ contract MixinTransfer is
     onlyKeyOwnerOrApproved(_tokenId)
   {
     require(_recipient != address(0), 'INVALID_ADDRESS');
-    uint fee = getTransferFee(_from);
+    uint fee = getTransferFee(_from, 0);
 
     Key storage fromKey = keyByOwner[_from];
     Key storage toKey = keyByOwner[_recipient];
+    uint id = fromKey.tokenId;
 
     uint previousExpiration = toKey.expirationTimestamp;
+    // subtract the fee from the senders key before the transfer
+    _timeMachine(id, fee, false);
```

Comment:
Looking at this again, should we maybe add an event inside timeMachine to emit the fee deducted?

Reply:
will do.